src/plugins/data_source_management/public/components/data_source_table/data_source_table.tsx

```diff
@@ -245,6 +252,22 @@ export const DataSourceTable = ({ history }: RouteComponentProps) => {
       });
   };
 
+  const setDefaultDataSource = async () => {
+    try {
+      for (const dataSource of selectedDataSources) {
+        if (uiSettings.get('defaultDataSource') === dataSource.id) {
+          await handleSetDefaultDatasourceAfterDeletion(savedObjects.client, uiSettings);
+        }
+      }
+    } catch (e) {
+      setIsLoading(false);
```

Comment:
what happens if error happens and we set isloading as false?

Reply:
It will stop loading and show error as "Unable to set the Data Source to default. Please try it again."

Reply:
Should use setIsDeleting(false), let me rephrase this part